File: app/src/main/java/com/example/shopmall/adapter/ChannelAdapter.java
package com.example.shopmall.adapter;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.net.Constant;
import com.example.shopmall.R;
import com.example.shopmall.bean.HomepageBean;

import java.util.List;

/**
 * Created by Administrator on 2016/9/28.
 */
public class ChannelAdapter extends BaseAdapter {
    private Context mContext;
    private List<HomepageBean.ResultBean.ChannelInfoBean> channel_info;

    public ChannelAdapter(Context mContext, List<HomepageBean.ResultBean.ChannelInfoBean> channel_info) {
        this.mContext = mContext;
        this.channel_info = channel_info;
    }


    @Override
    public int getCount() {
        return channel_info == null ? 0 : channel_info.size();
    }

    @Override
    public Object getItem(int position) {
        return channel_info.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder viewHolder;
        if (convertView == null) {
            convertView = View.inflate(mContext, R.layout.item_channel, null);
            viewHolder = new ViewHolder();
            viewHolder.ivChannel = convertView.findViewById(R.id.iv_channel);
            viewHolder.tvChannel = convertView.findViewById(R.id.tv_channel);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) convertView.getTag();
        }

        HomepageBean.ResultBean.ChannelInfoBean channelInfoBean = channel_info.get(position);
        viewHolder.tvChannel.setText(channelInfoBean.getChannel_name());
        Glide.with(mContext)
                .load(Constant.BASE_URL_IMAGE +channelInfoBean.getImage())
                .into(viewHolder.ivChannel);
        return convertView;
    }

    class ViewHolder {
        ImageView ivChannel;
        TextView tvChannel;
    }
}

File: app/src/main/java/com/example/dimensionleague/home/adapter/ChannelAdapter.kt
package com.example.dimensionleague.home.adapter


import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import com.bumptech.glide.Glide
import com.example.dimensionleague.R
import com.example.common.HomeBean
import com.example.net.AppNetConfig
import kotlinx.android.synthetic.main.home_channel_item.view.*

class ChannelAdapter(
    channelInfo: List<HomeBean.ResultBean.ChannelInfoBean>
) : BaseAdapter() {
    private var channelInfo:List<HomeBean.ResultBean.ChannelInfoBean>? = null
    init {
        this.channelInfo = channelInfo
    }
    override fun getView(position: Int, view: View?, parent: ViewGroup?): View {
        val views:View =
            LayoutInflater.from(parent!!.context).inflate(R.layout.home_channel_item,parent,false)
        lateinit var holder:ViewHolder
        holder = ViewHolder(views)
        views.tag = holder
        Glide.with(parent.context).load("${AppNetConfig.BASE_URl_IMAGE}${channelInfo!![position].image}").into(holder.ivChanner)
        holder.tvChanner.text = channelInfo!![position].channel_name
        return views
    }

    override fun getItem(position: Int): Any {
        return channelInfo!![position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getCount(): Int {
        return channelInfo!!.size
    }

    inner class ViewHolder(itemView: View){
        var ivChanner = itemView.channel_iv!!
        var tvChanner = itemView.channel_tv!!
    }
}